app/models/user.rb

```diff
@@ -591,6 +595,10 @@ def cohorts
     portal_teacher ? portal_teacher.cohorts : (portal_student ? portal_student.cohorts : [])
   end
 
+  def cohort_projects
+    cohorts.map {|c| c.project}.flatten.compact.uniq
```

Comment:
Hm, but `cohorts` is a method, not association. There is no direct association between `User` and `Cohort`, cohorts are inferred from `portal_teacher` or `portal_student` instances. Student's cohorts are actually inferred from his teachers' cohorts, teachers are inferred from classes etc., and most of that is implemented as methods, not associations. Maybe it's possible to change everything to associations. At the end we would need to define method anyway, like:
```ruby
def cohort_projects
  teacher_cohort_projects || student_cohort_projects
end
```
I can try.